gg-fleet-statusd/src/fleet_status_service.c

```diff
@@ -167,36 +167,44 @@ GglError publish_fleet_status_update(
         static uint8_t version_resp_mem[128] = { 0 };
         GglBuffer version_resp = GGL_BUF(version_resp_mem);
         ret = ggl_gg_config_read_str(
-            GGL_BUF_LIST(GGL_STR("services"), pair->key, GGL_STR("version")),
+            GGL_BUF_LIST(
+                GGL_STR("services"), component->buf, GGL_STR("version")
+            ),
             &version_resp
         );
-
         if (ret != GGL_ERR_OK) {
             GGL_LOGE(
-                "Unable to retrieve version of %s. Cannot publish fleet status "
-                "update.",
-                pair->key.data
+                "Unable to retrieve version of %.*s. Cannot publish fleet "
+                "status update.",
+                (int) component->buf.len,
+                component->buf.data
             );
             return GGL_ERR_NOENTRY;
         }
-
         ret = ggl_buf_clone(version_resp, &balloc.alloc, &version_resp);
         if (ret != GGL_ERR_OK) {
             GGL_LOGE("Failed to copy version response buffer.");
             return ret;
         }
 
         // retrieve component health status
-        uint8_t component_health_arr[NAME_MAX];
-        GglBuffer component_health = GGL_BUF(component_health_arr);
-        ret = retrieve_component_health_status(pair->key, &component_health);
+        static uint8_t component_health_mem[NAME_MAX];
+        GglBuffer component_health = GGL_BUF(component_health_mem);
+        ret = retrieve_component_health_status(
+            component->buf, &component_health
+        );
         if (ret != GGL_ERR_OK) {
             // skip reporting for components we fail to retrieve health statuses
-            // for
+            // for.
+            // TODO: Should all failures in this loop cause an error or warning
+            // and continue instead of return? because we want to update the
+            // status of any components we can, without one component affecting
+            // others
```

Comment:
Agreed we should still publish updates for components that don't fail. Can that be included in this pr? If not we can leave this and come back to it

Reply:
Added